llama/llama.go

```diff
@@ -579,17 +579,19 @@ func SchemaToGrammar(schema []byte) []byte {
 	cStr := C.CString(string(schema))
 	defer C.free(unsafe.Pointer(cStr))
 
-	// Allocate buffer for grammar based on schema length but with upper bound
-	maxLen := min(1024*1024, len(schema)*4)
```

Comment:
I think we should still have some upper bound to @jmorganca's point. We should be able to safely put this limit at 2MB and the `len(schema*4)` can stay.